c10/core/TensorImpl.h

```diff
@@ -1492,7 +1492,7 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
   void refresh_contiguous() {
     is_contiguous_ = compute_contiguous();
     is_channels_last_contiguous_ = compute_channels_last_contiguous();
-    is_channels_last_ = is_channels_last_contiguous_ || compute_strides_like_channels_last();
+    is_channels_last_ = compute_strides_like_channels_last();
```

Comment:
what happened here? It's always good to take shortcut if available.

Reply:
Except this shortcut is not the right thing to do.

`is_channels_last_contiguous_` is not equivalent to `is_channels_last_`.
In ambiguous cases `NC11`, where the tensor has both `is_contiguous_` and `is_channels_last_contiguous_` set as true. We need to use `is_channels_last_` to represent the desired `memory_format`.

I'll put a comment here.